ush/python/pygw/src/pygw/task.py

```diff
@@ -0,0 +1,45 @@
+class Task:
+    """
+    Base class for all tasks
+    """
+
+    def __init__(self, config, *args, **kwargs):
+        """
+        Every task needs a config.
+        Additional arguments (or key-value arguments) can be provided.
+
+        Parameters
+        ----------
+        config : Dict
+                 dictionary object containing task configuration
+
+        *args : tuple
+                Additional arguments to `Task`
+
+        **kwargs : dict, optional
+                   Extra keyword arguments to `Task`
+        """
+
+        # Store the config and arguments as attributes of the object
+        self.config = config
+
+        for arg in args:
+            setattr(self, str(arg), arg)
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def initialize(self):
```

Comment:
Short descriptions with what each of these methods should be used for in the child classes?